packages/ra-ui-materialui/src/list/SingleFieldList.tsx

```diff
@@ -159,14 +159,17 @@ export const SingleFieldListClasses = {
 const Root = styled('div', {
     name: PREFIX,
     overridesResolver: (props, styles) => styles.root,
-})({
+})(({ theme }) => ({
     display: 'flex',
     flexWrap: 'wrap',
 
     [`& .${SingleFieldListClasses.link}`]: {
         textDecoration: 'none',
+        '& > *': {
+            color: theme.palette.primary.main,
```

Comment:
This is the fix - and the indication of how to revert this change in user land with a custom theme, if your requirements lead you in the opposite direction.